front/src/routes/integration/all/ewelink/setup-page/index.js

```diff
@@ -0,0 +1,33 @@
+import { Component } from 'preact';
+import { connect } from 'unistore/preact';
+import actions from '../actions';
+import EweLinkPage from '../EweLinkPage';
+import SetupTab from './SetupTab';
+import { WEBSOCKET_MESSAGE_TYPES } from '../../../../../../../server/utils/constants';
+
+@connect(
+  'user,session,eweLinkUsername,eweLinkPassword,connectEweLinkStatus,eweLinkConnected,eweLinkConnectionError',
+  actions
+)
+class EweLinkSetupPage extends Component {
+  componentWillMount() {
+    this.props.getIntegrationByName('ewelink');
+    this.props.loadProps();
+    this.props.session.dispatcher.addListener(WEBSOCKET_MESSAGE_TYPES.EWELINK.CONNECTED, () =>
```

Comment:
If you add a listener you need to remove the listener on `componentWillUnmount`. 

See for example: https://github.com/GladysAssistant/Gladys/blob/da9fc1e70a11358b5c52bd6259ff12301b6bb484/front/src/routes/integration/all/zwave/node-operation-page/index.js#L75

Reply:
I think there is a mistake L78 in this example...

Reply:
indeed! Nice catch :D